pkg/config/analysis/incluster/controller.go

```diff
@@ -0,0 +1,125 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package incluster
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	v1alpha12 "istio.io/api/analysis/v1alpha1"
+	"istio.io/api/meta/v1alpha1"
+	"istio.io/istio/pilot/pkg/config/kube/arbitraryclient"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/status"
+	"istio.io/istio/pkg/config/analysis"
+	"istio.io/istio/pkg/config/analysis/analyzers"
+	"istio.io/istio/pkg/config/analysis/diag"
+	"istio.io/istio/pkg/config/analysis/local"
+	"istio.io/istio/pkg/config/resource"
+	"istio.io/istio/pkg/config/schema"
+	"istio.io/istio/pkg/config/schema/collection"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/kube"
+	"istio.io/pkg/log"
+)
+
+// Controller manages repeatedly running analyzers in istiod, and reporting results
+// via istio status fields.
+type Controller struct {
+	analyzer  *local.IstiodAnalyzer
+	statusctl *status.Controller
+}
+
+func NewController(stop <-chan struct{}, rwConfigStore, configController model.ConfigStoreCache,
+	kubeClient kube.Client, namespace string, statusManager *status.Manager, domainSuffix string) (*Controller, error) {
+	ia := local.NewIstiodAnalyzer(schema.MustBuildMetadata(configController.Schemas()), analyzers.AllCombined(),
+		"", resource.Namespace(namespace), func(name collection.Name) {}, true)
+	ia.AddSource(rwConfigStore)
+	ctx := status.NewIstioContext(stop)
+	// TODO: many of the types in PilotGatewayAPI (watched above) are duplicated
+	// I'm not sure why, but we shouldn't watch them twice.
+	duplicates := []collection.Schema{}
+	for k := range analysis.ContainmentMapSchema(rwConfigStore.Schemas()) {
+		duplicates = append(duplicates, k)
+	}
+	store, err := arbitraryclient.NewForSchemas(ctx, kubeClient, "default",
+		domainSuffix, collections.All.Remove(duplicates...))
+	if err != nil {
+		return nil, fmt.Errorf("unable to load common types for analysis, releasing lease: %v", err)
+	}
+	ia.AddSource(store)
+	kubeClient.RunAndWait(stop)
+	err = ia.Init(stop)
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize analysis controller, releasing lease: %s", err)
+	}
+	ctl := statusManager.CreateIstioStatusController(func(status *v1alpha1.IstioStatus, context interface{}) *v1alpha1.IstioStatus {
+		msgs := context.(diag.Messages)
+		// zero out analysis messages, as this is the sole controller for those
+		status.ValidationMessages = []*v1alpha12.AnalysisMessageBase{}
+		for _, msg := range msgs {
+			status.ValidationMessages = append(status.ValidationMessages, msg.AnalysisMessageBase())
+		}
+		return status
+	})
+	return &Controller{analyzer: ia, statusctl: ctl}, nil
+}
+
+// Run is blocking
+func (c *Controller) Run(stop <-chan struct{}) {
+	t := time.NewTicker(10 * time.Second)
```

Comment:
I think it is fine for now. Maybe add a todo to trigger with watcher?

Reply:
todo's are too easily lost.  I created a follow-up issue. https://github.com/istio/istio/issues/36101